File: trunk/sources/net.sf.j2s.lib/j2slib/net/sf/j2s/ajax/AJAXVariableInitializer.js
$_J("net.sf.j2s.ajax");
Clazz.load(["org.eclipse.jdt.core.ClasspathVariableInitializer"],["net.sf.j2s.ajax.AjaxPlugin","org.eclipse.core.resources.ResourcesPlugin","org.eclipse.core.runtime.NullProgressMonitor","$.Path","$.Platform","org.eclipse.jdt.core.JavaCore","org.eclipse.jdt.internal.launching.LaunchingPlugin"],function(){
c$=$_C(function(){
this.fMonitor=null;
$_Z(this,arguments);
},net.sf.j2s.ajax,"AJAXVariableInitializer",org.eclipse.jdt.core.ClasspathVariableInitializer);
$_V(c$,"initialize",
function(variable){
var newPath=null;
var starterURL=net.sf.j2s.ajax.AjaxPlugin.getDefault().getBundle().getEntry("/"+java.io.File.separator);
var root=".";
try{
root=org.eclipse.core.runtime.Platform.asLocalURL(starterURL).getFile();
}catch(e){
if($_O(e,java.io.IOException)){
e1.printStackTrace();
}else{
throw e;
}
}
if("AJAX_CORE".equals(variable)){
newPath=org.eclipse.core.runtime.Path.fromPortableString(root+"/ajaxcore.jar");
}else if("AJAX_SWT".equals(variable)){
newPath=org.eclipse.core.runtime.Path.fromPortableString(root+"/ajaxswt.jar");
}else if("AJAX_CORE_SRC".equals(variable)){
newPath=org.eclipse.core.runtime.Path.fromPortableString(root+"/ajaxcoresrc.zip");
}else if("AJAX_SWT_SRC".equals(variable)){
newPath=org.eclipse.core.runtime.Path.fromPortableString(root+"/ajaxswtsrc.zip");
}if(newPath==null){
return;
}var workspace=org.eclipse.core.resources.ResourcesPlugin.getWorkspace();
var wsDescription=workspace.getDescription();
var wasAutobuild=wsDescription.isAutoBuilding();
try{
this.setAutobuild(workspace,false);
this.setJREVariable(newPath,variable);
}catch(e){
if($_O(e,org.eclipse.core.runtime.CoreException)){
org.eclipse.jdt.internal.launching.LaunchingPlugin.log(ce);
return;
}else{
throw e;
}
}finally{
try{
this.setAutobuild(workspace,wasAutobuild);
}catch(e){
if($_O(e,org.eclipse.core.runtime.CoreException)){
org.eclipse.jdt.internal.launching.LaunchingPlugin.log(ce);
}else{
throw e;
}
}
}
},"~S");
$_M(c$,"setJREVariable",
($fz=function(newPath,$var){
org.eclipse.jdt.core.JavaCore.setClasspathVariable($var,newPath,this.getMonitor());
},$fz.isPrivate=true,$fz),"org.eclipse.core.runtime.IPath,~S");
$_M(c$,"setAutobuild",
($fz=function(ws,newState){
var wsDescription=ws.getDescription();
var oldState=wsDescription.isAutoBuilding();
if(oldState!=newState){
wsDescription.setAutoBuilding(newState);
ws.setDescription(wsDescription);
}return oldState;
},$fz.isPrivate=true,$fz),"org.eclipse.core.resources.IWorkspace,~B");
$_M(c$,"getMonitor",
function(){
if(this.fMonitor==null){
return new org.eclipse.core.runtime.NullProgressMonitor();
}return this.fMonitor;
});
});

File: tags/net.sf.j2s.ajax_v2.0.2_stable.20140625/src/net/sf/j2s/ajax/AJAXVariableInitializer.java
/*******************************************************************************
 * Copyright (c) 2007 java2script.org and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     Zhou Renjian - initial API and implementation
 *******************************************************************************/

package net.sf.j2s.ajax;

import java.io.File;
import java.io.IOException;
import java.net.URL;
import org.eclipse.core.resources.IWorkspace;
import org.eclipse.core.resources.IWorkspaceDescription;
import org.eclipse.core.resources.ResourcesPlugin;
import org.eclipse.core.runtime.CoreException;
import org.eclipse.core.runtime.FileLocator;
import org.eclipse.core.runtime.IPath;
import org.eclipse.core.runtime.IProgressMonitor;
import org.eclipse.core.runtime.IStatus;
import org.eclipse.core.runtime.NullProgressMonitor;
import org.eclipse.core.runtime.Path;
import org.eclipse.core.runtime.Status;
import org.eclipse.jdt.core.ClasspathVariableInitializer;
import org.eclipse.jdt.core.JavaCore;

/**
 * @author zhou renjian
 *
 * 2006-6-5
 */
public class AJAXVariableInitializer extends ClasspathVariableInitializer {


	/**
	 * The monitor to use for progress reporting.
	 * May be null
	 */
	private IProgressMonitor fMonitor;
	
	/**
	 * @see ClasspathVariableInitializer#initialize(String)
	 */
	public void initialize(String variable) {
		IPath newPath= null;
        URL starterURL = AjaxPlugin.getDefault().getBundle().getEntry(File.separator);
		String root = "."; //$NON-NLS-1$
		try {
			root = FileLocator.toFileURL(starterURL).getFile();
		} catch (IOException e1) {
			e1.printStackTrace();
		}
		if ("AJAX_CORE".equals(variable)) {
			newPath = Path.fromPortableString(root + "/ajaxcore.jar");
		} else if ("AJAX_SWT".equals(variable)) {
			newPath = Path.fromPortableString(root + "/ajaxswt.jar");
		} else if ("AJAX_RPC".equals(variable)) {
			newPath = Path.fromPortableString(root + "/ajaxrpc.jar");
		} else if ("AJAX_PIPE".equals(variable)) {
			newPath = Path.fromPortableString(root + "/ajaxpipe.jar");
		} else if ("AJAX_STORE".equals(variable)) {
			newPath = Path.fromPortableString(root + "/ajaxstore.jar");
		} else if ("J2S_ANNOTATION".equals(variable)) {
			newPath = Path.fromPortableString(root + "/j2stag.jar");
		} else if ("AJAX_CORE_SRC".equals(variable)) {
			newPath = Path.fromPortableString(root + "/ajaxcoresrc.zip");
		} else if ("AJAX_SWT_SRC".equals(variable)) {
			newPath = Path.fromPortableString(root + "/ajaxswtsrc.zip");
		} else if ("AJAX_RPC_SRC".equals(variable)) {
			newPath = Path.fromPortableString(root + "/ajaxrpcsrc.zip");
		} else if ("AJAX_PIPE_SRC".equals(variable)) {
			newPath = Path.fromPortableString(root + "/ajaxpipesrc.zip");
		} else if ("AJAX_STORE_SRC".equals(variable)) {
			newPath = Path.fromPortableString(root + "/ajaxstoresrc.zip");
		} else if ("J2S_ANNOTATION_SRC".equals(variable)) {
			newPath = Path.fromPortableString(root + "/j2stagsrc.zip");
		}

		if (newPath == null) {
			return ;
		}
//		IVMInstall vmInstall= JavaRuntime.getDefaultVMInstall();
//		if (vmInstall != null) {
			IWorkspace workspace= ResourcesPlugin.getWorkspace();
			IWorkspaceDescription wsDescription= workspace.getDescription();				
			boolean wasAutobuild= wsDescription.isAutoBuilding();
			try {
				setAutobuild(workspace, false);
				setJREVariable(newPath, variable);	
			} catch (CoreException ce) {
				AjaxPlugin.getDefault().getLog().log(new Status(IStatus.ERROR,
						"net.sf.j2s.ajax", IStatus.ERROR, ce.getMessage(), ce));
				return;
			} finally {
				try {
					setAutobuild(workspace, wasAutobuild);
				} catch (CoreException ce) {
					AjaxPlugin.getDefault().getLog().log(new Status(IStatus.ERROR,
							"net.sf.j2s.ajax", IStatus.ERROR, ce.getMessage(), ce));
				}
			}
//		}
	}
	
	private void setJREVariable(IPath newPath, String var) throws CoreException {
		JavaCore.setClasspathVariable(var, newPath, getMonitor());
	}
	
	private boolean setAutobuild(IWorkspace ws, boolean newState) throws CoreException {
		IWorkspaceDescription wsDescription= ws.getDescription();
		boolean oldState= wsDescription.isAutoBuilding();
		if (oldState != newState) {
			wsDescription.setAutoBuilding(newState);
			ws.setDescription(wsDescription);
		}
		return oldState;
	}
	
	protected IProgressMonitor getMonitor() {
		if (fMonitor == null) {
			return new NullProgressMonitor();
		}
		return fMonitor;
	}
	
}
